swagger.yaml

```diff
@@ -107,265 +243,979 @@ paths:
+      parameters:
+      - $ref: '#/parameters/id'
+      - $ref: '#/parameters/new_version'
+      responses:
+        201:
+          description: "A json object containing a version"
+          schema:
+            $ref: '#/definitions/NewVersionResponse'
+        400:
+          description: "Invalid request body"
+        401:
+          description: "Unauthorised to update version of dataset"
+        403:
+          description: "Forbidden to overwrite version of dataset, already published"
+        404:
+          description: "No edition was found for a dataset using the id and edition provided"
+        500:
+          $ref: '#/responses/InternalError'
     get:
       tags:
-      - "Dataset"
+      - "Public"
       summary: "Get a specific version and edition of a dataset"
       description: "Get a specific version and edition of a dataset"
       parameters:
-      - $ref: '#/parameters/id'
       - $ref: '#/parameters/edition'
+      - $ref: '#/parameters/id'
       - $ref: '#/parameters/version'
       responses:
         200:
           description: "A json object containing the edition and version of a dataset"
           schema:
             $ref: '#/definitions/Version'
+        401:
+          $ref: '#/responses/UnauthorisedError'
         404:
-          description: "No dataset edition with the specific version was found using the id, edition and version provided"
+          description: "No version was found for an edition of a dataset using the id, edition and version provided"
         500:
-          description: "Failed to process the request due to an internal error"
+          $ref: '#/responses/InternalError'
   /datasets/{id}/editions/{edition}/versions/{version}/dimensions:
     get:
       tags:
-      - "Dataset"
+      - "Public"
       summary: "Get all dimensions from a dataset"
       description: "Get all dimensions which are used in the dataset"
       parameters:
-      - $ref: '#/parameters/id'
       - $ref: '#/parameters/edition'
+      - $ref: '#/parameters/id'
       - $ref: '#/parameters/version'
       responses:
         200:
           description: "A json list of dimensions"
           schema:
             $ref: '#/definitions/Dimensions'
+        401:
+          $ref: '#/responses/UnauthorisedError'
         404:
-          description: "No dataset edition with the specific version was found using the id, edition and version provided"
+          description: "No version was found for an edition of a dataset using the id, edition and version provided"
         500:
-          description: "Failed to process the request due to an internal error"
-  /datasets/{id}/editions/{edition}/versions/{version}/dimensions/{dimension}:
+          $ref: '#/responses/InternalError'
+  /datasets/{id}/editions/{edition}/versions/{version}/dimensions/{dimension}/options:
     get:
       tags:
-      - "Dataset"
-      summary: "Get all values from a dimension"
```

Comment:
All references to dimension values or value should be updated to dimension options and option respectively.

Reply:
Update all dimension values references to dimension options